src/java/org/apache/cassandra/index/SecondaryIndexManager.java

```diff
@@ -1126,17 +1165,26 @@ public UpdateTransaction newUpdateTransaction(PartitionUpdate update, WriteConte
     {
         if (!hasIndexes())
             return UpdateTransaction.NO_OP;
-
-        Index.Indexer[] indexers = indexes.values().stream()
-                                          .map(i -> i.indexerFor(update.partitionKey(),
-                                                                 update.columns(),
-                                                                 nowInSec,
-                                                                 ctx,
-                                                                 IndexTransaction.Type.UPDATE))
-                                          .filter(Objects::nonNull)
-                                          .toArray(Index.Indexer[]::new);
-
-        return indexers.length == 0 ? UpdateTransaction.NO_OP : new WriteTimeTransaction(indexers);
+        
+        ArrayList<Index.Indexer> idxrs = new ArrayList<>();
```

Comment:
Ignorable nit: perhaps we could name this variable `indexers`, and use `idxrs` for the array below. Also, I wonder if it would make sense to use an iterable instead of an array in `WriteTimeTransaction`, so we can directly pass this `ArrayList` without the `toArray` conversion. Would it hurt performance?

Reply:
I was trying to get all this path without iterators which would be ideal, but `indexes.values()` is a `Collection` which I can't access via `get()`. The previous streams would have the same array conversion, which is suspicious as the immediate solution would be to use the `ArrayList`. Being the hot path maybe we want to keep the 'pure' array index access even though `ArrayLists`'s overhead might be minimal if anyting at all. Sounds good?